pkg/morph/client/client.go

```diff
@@ -220,10 +220,6 @@ func (c *Client) TestInvokeIterator(contract util.Uint160, method string, args .
 		return nil, err
 	}
 
-	if val.State != HaltState {
-		return nil, wrapNeoFSError(&notHaltStateError{state: val.State, exception: val.FaultException})
-	}
-
 	sid, iter, err := unwrap.SessionIterator(val, nil)
```

Comment:
And then this should just be `unwrap.SessionIterator(c.rpcActorCall(...))`, that's why it takes an error for input.